src/test/java/org/reaktivity/nukleus/maven/plugin/internal/generated/ListWithPhysicalAndLogicalLengthFW.java

```diff
@@ -0,0 +1,330 @@
+/**
+ * Copyright 2016-2019 The Reaktivity Project
+ *
+ * The Reaktivity Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package org.reaktivity.nukleus.maven.plugin.internal.generated;
+
+import java.nio.charset.StandardCharsets;
+
+import org.agrona.BitUtil;
+import org.agrona.DirectBuffer;
+import org.agrona.MutableDirectBuffer;
+import org.reaktivity.reaktor.internal.test.types.Flyweight;
+import org.reaktivity.reaktor.internal.test.types.StringFW;
+
+public class ListWithPhysicalAndLogicalLengthFW extends Flyweight
+{
+    public static final int PHYSICAL_LENGTH_OFFSET = 0;
+
+    private static final int PHYSICAL_LENGTH_SIZE = BitUtil.SIZE_OF_INT;
+
+    public static final int LOGICAL_LENGTH_OFFSET = PHYSICAL_LENGTH_OFFSET + PHYSICAL_LENGTH_SIZE;
+
+    private static final int LOGICAL_LENGTH_SIZE = BitUtil.SIZE_OF_INT;
+
+    public static final int BIT_MASK_OFFSET = LOGICAL_LENGTH_OFFSET + LOGICAL_LENGTH_SIZE;
+
+    private static final int BIT_MASK_SIZE = BitUtil.SIZE_OF_INT;
+
+    public static final int FIELD_OFFSET_FIELD0 = BIT_MASK_OFFSET + BIT_MASK_SIZE;
+
+    private static final int FIELD_SIZE_FIELD1 = BitUtil.SIZE_OF_INT;
+
+    private static final int FIELD_INDEX_FIELD0 = 0;
+
+    private static final int FIELD_INDEX_FIELD1 = 1;
+
+    private static final int FIELD_INDEX_FIELD2 = 2;
+
+    private final StringFW field0RO = new StringFW();
+
+    private final StringFW field2RO = new StringFW();
+
+    private final int[] optionalOffsets = new int[FIELD_INDEX_FIELD2 + 1];
+
+    public long length()
+    {
+        return buffer().getInt(offset() + LOGICAL_LENGTH_OFFSET) & 0xFFFF_FFFFL;
+    }
+
+    private long bitmask()
+    {
+        return buffer().getInt(offset() + BIT_MASK_OFFSET) & 0xFFFF_FFFFL;
```

Comment:
Given that the bitmask is only used internally, it probably doesn't need to be converted to long to represent unsigned, you can just use an int directly and still do all the field mask tests you need.